### PR TITLE
fix: live preview: added variant support in data cslp tag function

### PR DESCRIPTION
**Pull request by contentstackMridul**
fix: live preview: added variant support in data cslp tag function

### CODE REVIEW

File: src/entry-editable.ts [hunk 1]
```diff
@@ -1,105 +1,122 @@
...
-    const tags: any = {}
-    Object.entries(content).forEach(([key, value]) => {      
-        if (key === '$') return
+  export function addTags(entry: EntryModel, contentTypeUid: string, tagsAsObject: boolean, locale: string = 'en-us') : void {  
+      if (entry) { 
+          const appliedVariants = entry._applied_variants || null;
+          entry.$ = getTag(entry, `${contentTypeUid}.${entry.uid}.${locale}`, tagsAsObject, locale, {_applied_variants: appliedVariants, shouldApplyVariant: !!appliedVariants, metaKey: ''})      }  
+  }
+  
+  function getTag(content: object, prefix: string, tagsAsObject: boolean, locale: string, appliedVariants: {_applied_variants:{[key: string]: any}, shouldApplyVariant: boolean, metaKey: string}): object {
```
Comment (KANE-99) [inline src/entry-editable.ts:9]: this should be optional argument?

File: src/entry-editable.ts [hunk 2]
```diff
@@ -1,105 +1,122 @@
...
+      const tags: any = {}
+      const { metaKey, shouldApplyVariant, _applied_variants } = appliedVariants
```
Comment (KANE-99) [inline src/entry-editable.ts:11]: handle case when appliedVariants is not there

File: src/entry-editable.ts [hunk 3]
```diff
@@ -1,105 +1,122 @@
...
+      Object.entries(content).forEach(([key, value]) => {      
+          if (key === '$') return
+          let metaUID = value && typeof value === 'object' && value._metadata && value._metadata.uid ? value._metadata.uid : '';
+          let updatedMetakey = appliedVariants.shouldApplyVariant ? `${appliedVariants.metaKey ? appliedVariants.metaKey + '.' : '' }${key}` : '';
```
Comment (KANE-99) [inline src/entry-editable.ts:15]: `appliedVariants?.shouldApplyVariant`

**Review by KANE-99 — APPROVED** [review]
LGTM

**Review by nadeem-cs — APPROVED** [review]

**Pull request closed without merging**